### ISSUE TITLE
Multiselect: The implementation confuses/mislabel the title, label

### ISSUE THREAD

**Issue by alisonjoseph** [carbon-design-system/carbon#13096]
The implementation confuses/mislabel the title, label,
The multi-select markedly departs from the terminology of Dropdown. First it swaps out the "label" text preceding the input and calls it titleText. You can see the "This is MultiSelect Title" specified with the value in this screen shot. The instructions _do_ say "Provide text to be used in a `label` element..." but it is singularly unhelpful to change the terminology.

![image](https://user-images.githubusercontent.com/14298245/203642640-e3c3fe0b-e521-4f6d-b4d6-546b90c6c5a9.png)

The added confusion is made clear when we see what they have called the default option value, "label", here given a value of "This is a label". Note that the control description here -- "Generic `label`..." -- is _extremely_ confusing.

![image](https://user-images.githubusercontent.com/14298245/203640017-b1e2c186-40f7-4dc9-931e-882c412df46a.png)

This is also the case in the Live demo area; an attribute called "label" which should be the default option. As noted elsewhere, this default option _can_ have the effect of being a placeholder, but it also remains the top option in the list. 

<img width="599" alt="image" src="https://user-images.githubusercontent.com/14298245/211894416-c9bc9f6e-8134-4db5-bcdc-c15db78399eb.png">

**These variables should be renamed to match those in the Dropdown anatomy section.**
